src/test/rpc/LedgerRPC_test.cpp

```diff
@@ -0,0 +1,283 @@
+        env.fund(XRP(10000), alice);
+        env.close();
+
+        auto jrr = env.rpc("ledger_closed") [jss::result];
+        BEAST_EXPECT(jrr[jss::ledger_hash] == to_string(env.closed()->info().hash));
+        BEAST_EXPECT(jrr[jss::ledger_index] == 3);
+
+        Json::Value jvParams;
+        jvParams[jss::account_root] = alice.human();
+        jvParams[jss::ledger_hash] = jrr[jss::ledger_hash];
+        jrr = env.rpc ( "json", "ledger_entry", to_string(jvParams) ) [jss::result];
+        BEAST_EXPECT(jrr.isMember(jss::node));
+        BEAST_EXPECT(jrr[jss::node][jss::Account] == alice.human());
+        BEAST_EXPECT(jrr[jss::node][sfBalance.fieldName] == XRP(10000).value().getText());
+    }
+
+    void testMalformedAccountRoot()
+    {
+        testcase("Malformed Ledger Entry Request");
+        using namespace test::jtx;
+        Env env {*this};
+        Account const alice {"alice"};
+        env.fund(XRP(10000), alice);
+        env.close();
+
+        auto jrr = env.rpc("ledger_closed") [jss::result];
+
+        Json::Value jvParams;
+        jvParams[jss::account_root] = std::string(alice.human()).replace(0, 2, 2, 'x');
+        jvParams[jss::ledger_hash] = jrr[jss::ledger_hash];
+        jrr = env.rpc ( "json", "ledger_entry", to_string(jvParams) ) [jss::result];
+        BEAST_EXPECT(jrr[jss::error] == "malformedAddress");
+        BEAST_EXPECT(jrr[jss::status] == "error");
+        BEAST_EXPECT(jrr[jss::error_message] == Json::nullValue);
+    }
+
+    void testNotFoundAccountRoot()
+    {
+        testcase("Ledger Entry Not Found");
+        using namespace test::jtx;
+        Env env {*this};
+        Account const alice {"alice"};
+        env.fund(XRP(10000), alice);
+        env.close();
+
+        auto jrr = env.rpc("ledger_closed") [jss::result];
+
+        Json::Value jvParams;
+        jvParams[jss::account_root] = Account("bob").human();
+        jvParams[jss::ledger_hash] = jrr[jss::ledger_hash];
+        jrr = env.rpc ( "json", "ledger_entry", to_string(jvParams) ) [jss::result];
+        BEAST_EXPECT(jrr[jss::error] == "entryNotFound");
+        BEAST_EXPECT(jrr[jss::status] == "error");
+        BEAST_EXPECT(jrr[jss::error_message] == Json::nullValue);
+    }
+
+    void testAccountRootFromIndex()
+    {
+        testcase("Ledger Entry Request From Index");
+        using namespace test::jtx;
+        Env env {*this};
+        Account const alice {"alice"};
+        env.fund(XRP(10000), alice);
+        env.close();
+
+        auto jrr = env.rpc("ledger_closed") [jss::result];
+        BEAST_EXPECT(jrr[jss::ledger_hash] == to_string(env.closed()->info().hash));
+        BEAST_EXPECT(jrr[jss::ledger_index] == 3);
+
+        Json::Value jvParams;
+        jvParams[jss::account_root] = alice.human();
+        jvParams[jss::ledger_hash] = jrr[jss::ledger_hash];
+        jrr = env.rpc ( "json", "ledger_entry", to_string(jvParams) ) [jss::result];
+        jvParams[jss::index] = jrr[jss::index];
```

Comment:
I got away with that because of the order the params are checked/extracted in the handler (index is checked first), but yeah don't want to rely on that - will fix